src/days-2-4-adv/adv.py

```diff
@@ -91,32 +105,59 @@ def screen_message():
         print(element)
 
 
+# TODO: Simplify/Abstract this
+# Check if the item exists in the Items dict, and if exists,
+# Use the player pickup method to pickup the item, if not,
+# return a message printing item not found
+def item_exists(item):
+    bl = False
+    for itm in items:
+        if itm == item.capitalize():
+            bl = True
+            return True
+
+    if bl is False:
+        return False
+
+
 # START GAME
 # Initialize the character input, display the room message and initialize the player's input
 # These three functions run once then the while loop takes over
+
+
 set_init_player()
 screen_message()
 player_input()
 
 # While the input is not q, keep get game going
 while not player_inp == 'q':
+
+    # TODO: Abstract the command handler
     # if player chooses North
     if player_inp == 'n':
-        player.move_north()
+        player.movedir('north')
     elif player_inp == 's':
-        player.move_south()
+        player.movedir('south')
     elif player_inp == 'e':
-        player.move_east()
+        player.movedir('east')
     elif player_inp == 'w':
-        player.move_west()
+        player.movedir('west')
+    elif player_inp == 'look around':
+        player.look_around()
+    elif player_inp[:6] == 'pickup':
```

Comment:
I like this! This is a very creative way of implementing a pickup command. It might interest you though to know that you could use a certain combination of "strings and arrays" to implement complex commands like "pickup" and "look around". I don't want to spoil anything though. I see you have Abstract the command handler on your todo list!